client/rest-high-level/src/main/java/org/elasticsearch/client/dataframe/transforms/pivot/DateHistogramGroupSource.java

```diff
@@ -168,4 +168,55 @@ public boolean equals(Object other) {
     public int hashCode() {
         return Objects.hash(field, interval, dateHistogramInterval, timeZone, format);
     }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+
+        private String field;
+        private long interval = 0;
+        private DateHistogramInterval dateHistogramInterval;
+        private String format;
+        private ZoneId timeZone;
+
+        public Builder setField(String field) {
+            this.field = field;
+            return this;
+        }
+
+        public Builder setInterval(long interval) {
```

Comment:
Should this method accept "Duration" instead of "long"?

Reply:
I was trying to make this parallel the `DateHistogramAggregationBuilder`. Though I should add docs stating what time resolution `interval` is.

Reply:
Yeah, if we talk about usability here, it is usually error-prone if the user has to provide "long" but does not know what the unit is.

Reply:
@przemekwitek I added some docs and a new method that accepts a `TimeValue` parameter as well.